From 7dd62fe9c6520d1a2ee98bade664698031ed0914 Mon Sep 17 00:00:00 2001
Message-ID: <7dd62fe9c6520d1a2ee98bade664698031ed0914.1736059918.git.divya@subvertising.org>
In-Reply-To: <cover.1736059918.git.divya@subvertising.org>
References: <cover.1736059918.git.divya@subvertising.org>
From: Divya Ranjan <divya@subvertising.org>
Date: Sat, 4 Jan 2025 20:30:57 +0000
Subject: [PATCH 02/44] gnu: Add rust-serial-windows-0.4.

* gnu/packages/crates-io.scm (rust-serial-windows-0.4): New variable.

Change-Id: I979f5a1e83197dd6661413e2fa5f412f71418b48
---
 gnu/packages/crates-io.scm | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/gnu/packages/crates-io.scm b/gnu/packages/crates-io.scm
index c0ba7a3013..327cec42bb 100644
--- a/gnu/packages/crates-io.scm
+++ b/gnu/packages/crates-io.scm
@@ -70921,6 +70921,27 @@ (define-public rust-serial-test-derive-0.1
        (("rust-quote" ,rust-quote-0.6)
         ("rust-syn" ,rust-syn-0.15))))))
 
+(define-public rust-serial-windows-0.4
+  (package
+    (name "rust-serial-windows")
+    (version "0.4.0")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (crate-uri "serial-windows" version))
+       (file-name (string-append name "-" version ".tar.gz"))
+       (sha256
+        (base32 "0ql1vjy57g2jf218bhmgr98i41faq0v5vzdx3g9payi6fsvx7ihm"))))
+    (build-system cargo-build-system)
+    (arguments
+     `(#:cargo-inputs (("rust-libc" ,rust-libc-0.2)
+                       ("rust-serial-core" ,rust-serial-core-0.4))))
+    (home-page "https://github.com/dcuddeback/serial-rs")
+    (synopsis "Serial port implementation for Windows")
+    (description
+     "This package provides Serial port implementation for Windows.")
+    (license license:expat)))
+
 (define-public rust-servo-arc-0.3
   (package
     (name "rust-servo-arc")
-- 
2.47.1

